src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode;
+        if (first == null) {
+            newNode = new Node<>(null, value, null);
+            first = newNode;
+        } else {
+            newNode = new Node<>(last, value, null);
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
Try to reduce you _else if_ stetaments

Reply:
Sorry, I also think that the code that is needed only in one place should not be broken into pieces. In my opinion, calling a method inside another method is justified when it solves the problem of code duplication. Also, if the method grows to crazy proportions. Here I see no need to reduce the size of the method and move parts to other methods.

Reply:
asked about it in questions.